packages/docs/src/routes/docs/(qwik)/components/overview/index.mdx

```diff
@@ -113,40 +113,46 @@ export default component$(() => {
 
 ## Props
 
-Props are used to pass data from the parent into the component. Props are accessible via the `props` argument to the component$ function.
+Props are used to pass data from the parent into the component. They are accessible via the `props` argument to the `component$` function. Data passed as props is immutable, and cannot be changed. In cases where data sent as props needs to be updated, it is recommended to use signals.
```

Comment:
This is not actually the case. The data is mutable, but unless you use signals or stores, there won't be a reactive change. 
The props object itself is private to the component, it can also be mutated without problems, which might be handy when using props spreading.

Reply:
@codyroberts maybe changing "immutable" to "non reactive" would be more accurate in that case?

@wmertens can you give a short example of the props spreading benefit please?

Reply:
I tested it locally and I am still getting `Error: Code(17): props are immutable` when doing something simple like:

```js
export const Item = component$<ItemProps>((props) => {
  props.price = 10;
  return (
    <ul>
      <li>name: {props.name}</li>
      <li>quantity: {props.quantity}</li>
      <li>description: {props.description}</li>
      <li>price: {props.price}</li>
    </ul>
  );
});
```

I was making this change to the documentation in response to this thread:
https://github.com/QwikDev/qwik/issues/6423

Reply:
@codyroberts ah indeed you're right, I didn't notice that before. So indeed the props object is shallowly immutable, but not deeply. This does not error:

```tsx
import { component$ } from '@builder.io/qwik';

export const Foo = component$((props: any) => {
  // mutate a given prop
  props.bar.hello = 3

  return <button {...props}>A button</button>
})

export default component$(() => {
  return <Foo bar={{ hi: true }} />;
});
```

So deleting props isn't possible, you need to override via the props order.